frontend/src/pages/TracesExplorer/Filter/DurationSection.tsx

```diff
@@ -0,0 +1,143 @@
+import { Input, Slider } from 'antd';
+import { SliderRangeProps } from 'antd/es/slider';
+import { getMs } from 'container/Trace/Filters/Panel/PanelBody/Duration/util';
+import useDebouncedFn from 'hooks/useDebouncedFunction';
+import {
+	ChangeEventHandler,
+	Dispatch,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+
+import { addFilter, FilterType, traceFilterKeys } from './filterUtils';
+
+interface DurationProps {
+	selectedFilters: FilterType | undefined;
+	setSelectedFilters: Dispatch<SetStateAction<FilterType | undefined>>;
+}
+
+export function DurationSection(props: DurationProps): JSX.Element {
+	const { setSelectedFilters, selectedFilters } = props;
+
+	const getDuration = useMemo(() => {
+		if (selectedFilters?.durationNanoMin || selectedFilters?.durationNanoMax) {
+			return {
+				minDuration: selectedFilters?.durationNanoMin?.values || '',
+				maxDuration: selectedFilters?.durationNanoMax?.values || '',
+			};
+		}
+
+		if (selectedFilters?.durationNano) {
+			return {
+				minDuration: getMs(selectedFilters?.durationNano?.values?.[0] || ''),
+				maxDuration: getMs(selectedFilters?.durationNano?.values?.[1] || ''),
+			};
+		}
+
+		return {
+			maxDuration: '',
+			minDuration: '',
+		};
+	}, [selectedFilters]);
+
+	const [preMax, setPreMax] = useState<string>('');
+	const [preMin, setPreMin] = useState<string>('');
+
+	useEffect(() => {
+		setPreMax(getDuration.maxDuration as string);
+		setPreMin(getDuration.minDuration as string);
+	}, [getDuration]);
+
+	const updateDurationFilter = (min: string, max: string): void => {
+		const durationMin = 'durationNanoMin';
+		const durationMax = 'durationNanoMax';
+
+		addFilter(durationMin, min, setSelectedFilters, traceFilterKeys.durationNano);
+		addFilter(durationMax, max, setSelectedFilters, traceFilterKeys.durationNano);
+	};
+
+	const onRangeSliderHandler = (number: [string, string]): void => {
+		const [min, max] = number;
+
+		setPreMin(min);
+		setPreMax(max);
+	};
+
+	const debouncedFunction = useDebouncedFn(
+		(min, max) => {
+			updateDurationFilter(min as string, max as string);
+		},
+		1500,
+		undefined,
```

Comment:
?

Reply:
assigned - `options?: DebounceOptions` as undefined, though not assigning would also work, would cleanup in followup PR.